syntax = "proto3";

message Vector3 {
  float x = 1;
  float y = 2;
  float z = 3;
}

message Quaternion {
    float w = 1;
    float x = 2;
    float y = 3;
    float z = 4;
}

message Imu {
    double timestamp = 1;
    Quaternion orientation = 2;
    Vector3 angular_velocity = 3;
    Vector3 linear_acceleration = 4;
}